core/src/main/scala/org/ensime/indexer/ClassfileIndexer.scala

```diff
@@ -157,7 +158,7 @@ final class ClassfileIndexer(file: FileObject) extends SLF4JLogging {
 
     var clazz: RawClassfile
 
-    var internalRefs = new util.HashSet[FullyQualifiedName]()
+    var internalRefs = new util.HashSet[FullyQualifiedReference]()
```

Comment:
`HashSet` could be something faster, like a linked list. It is unlikely that we'll have dupes anymore, and the few that we do have we can filter later.